### DISCUSSION TITLE
I want to specify the SessionId of a specific user.

### DISCUSSION THREAD

**Discussion by lanopk** [dokan-dev/dokany#1056] [Ideas]
I want to specify the SessionId of the user.

Looking at the Dokan source code, if DOKAN_OPTION_CURRENT_SESSION is specified, GetCurrentSessionId is called and processed.

If you're running Dokan as a window service, you can't handle this.
So, instead of GetCurrentSessionId, it is better to use the user-specified sessionId.

For example
Specify "DOKAN_OPTION_USER_SESSIONID" for DOKAN_OPTION and
I wish I could specify "UserSessionId" (DWORD) in PDOKAN_OPTIONS.

thank you.

**Comment by Liryna** [comment]
Hi @lanopk ,

Unfortunately this is not currently possible. It would require changing the logic on the driver side to start a thread in the specified user context and process the mount in it.
It would be easier for the windows service to transfer the request to a process launched in the user session that does the mount.
